composite-base/src/Composite/Record.hs

```diff
@@ -82,6 +83,15 @@ instance Monad ((:->) s) where
   Val a >>= k = k a
   {-# INLINE (>>=) #-}
 
+instance NFData (s :-> a) where
+  rnf x = seq x ()
```

Comment:
this only reduces the `:->` constructor to normal form I think and doesn't recurse, I think you might want `rnf (Val x) = rnf x ()` instead?